### DISCUSSION TITLE
"fdroid" vs other release

### DISCUSSION THREAD

**Discussion by torgeros** [ImranR98/Obtainium#1372] [Q&A]
Hi everyone. For Obtainium, what is the difference between the "fdroid" apk file and the other one?

I know that apps often offer builds with different levels of tracking. Sometimes it is hard to find out, which version of the app you installed last time.

Does obtainium show somewhere, what the name of the last apk was? Obviously, the System-reported app version no longer shows which version you installed last time.

**Comment by inson1** [comment]
Also documentation in Readme about the difference would be great

**Comment by inson1** [comment]
@torgeros I think only opt-in app updater is different https://github.com/ImranR98/Obtainium/issues/30#issuecomment-1473447520 https://github.com/ImranR98/Obtainium/issues/30#issuecomment-1554971673

**Comment by Gropinou** [comment]
I was looking for the same thing and when digging in the source code, that's all I found.

**Comment by jeffmcneill** [comment]
I think the issue is about signatures or lack thereof. One cannot update using a different signature signed version. You'll get an error if you try.

**Reply by master-bob** [reply]
Yes that is correct for non-reproducible builds, where fdroid uses their own signing key. Reproducible builds on fdroid will use Obtainium's key so it shouldn't matter if you update with fdroid or gihub's apk.

**Comment by master-bob** [comment]
There is essentially no difference between the fdroid flavor and the normal flavor. Practically the only difference is the normal flavor adds Obtainium as an app within Obtainium, the fdroid flavor omits that.

You can see for yourself via [main_fdroid.dart](https://github.com/ImranR98/Obtainium/blob/main/lib/main_fdroid.dart#L2), [main.dart](https://github.com/ImranR98/Obtainium/blob/00d545b36ea9c2ff74f97a2a73d345771839bf00/lib/main.dart#L227), [build.sh](https://github.com/ImranR98/Obtainium/blob/main/build.sh#L52), and [release.yml](https://github.com/ImranR98/Obtainium/blob/main/.github/workflows/release.yml#L56). The building scripts (build.sh (local script) and release.yml (github building)) for the fdroid flavor call main_fdroid.dart, normal flavor is as usual.

**Comment by Dubz** [comment]
Found this from a search, but for some reason my instance of Obtanium on my secondary testing phone keeps asking whether I want the F-Droid variant or not whenever it has an update. Not sure how I can fix this to just always update without F-Droid. My primary phone never asks. Same exact model/OS. Don't know if I installed one differently than the other, but would be nice to know why it keeps occurring and how I can stop it. I would've thought the first prompt would've been it, not every time.
